api-lib/eas.ts

```diff
@@ -0,0 +1,78 @@
+import { EAS, SchemaEncoder } from '@ethereum-attestation-service/eas-sdk';
+import { Wallet } from 'ethers';
+import { getAddress } from 'ethers/lib/utils';
+
+import { Give } from '../_api/hasura/cron/giveOnchainSyncer';
+import { IN_PRODUCTION } from '../src/config/env';
+import { webAppURL } from '../src/config/webAppURL';
+import { baseChain } from '../src/features/cosoul/chains';
+import { coLinksPaths } from '../src/routes/paths';
+
+import { COSOUL_SIGNER_ADDR_PK } from './config';
+import { getProvider } from './provider';
+
+const EAS_CONTRACT_ADDR = '0x4200000000000000000000000000000000000021';
+const BASE_MAINNET_SCHEMA_UID =
+  '0x44ba9714fe382b86f0e126a592116bc2fc45905ee034d0e8b5db218bf366dafe';
+const BASE_SEPOLIA_SCHEMA_UID =
+  '0x55df90070189848b41fc4f2a8e07f82d6434718696286b9188ceb1d737f6cf87';
+
+export async function attestGiveOnchain(give: Give) {
+  const chainId = Number(baseChain.chainId);
+  const provider = getProvider(chainId);
+
+  const syncerWallet = new Wallet(COSOUL_SIGNER_ADDR_PK);
+  const signer = syncerWallet.connect(provider);
```

Comment:
do we want a diff addr for this or is this all good

Reply:
i think just this one and then we have onchain.coordinape.eth set to it